### ISSUE TITLE
Is RichEdit a necessity?

### ISSUE THREAD

**Issue by codernix** [fxarena/openfx-text#1]
Hey Andre,

Is RichEdit a must have or just to enrich the plugin?

Also, any ETA for when you will push the rest of the source code that will compile as an OFX?

**Comment by rodlie** [comment]
Got no ETA, but probably tonight.

**Comment by codernix** [comment]
On my end, I have tried the RichEdit and it didn't show anything, but as soon as I set the "Align" == "center", it showed and was wee tiny that I could barely notice.

For some reason, when I precede it by a Fusion Text+ node, it got a bit bigger.

https://ibb.co/n0Gs9Ww

The good news, Arabic is finally rendering correctly, since the inception of Fusion, so this is a first, that even Black Magic Design couldn't resolve for years.

The bad news, in the RichEdit panels don't have a way to change the fonts, color, size and what comes along, and no markup/tags worked.

So, I believe we're getting there very soon hopefully.

**Comment by rodlie** [comment]
Do not use RichTextOFX, it's a custom plugin for Natron, it will not work in other applications. In fact it does not work in Natron either at the moment, as it's a plugin that need custom support in the application (that I will add in Natron).

![screenshot](https://github.com/rodlie/openfx-text/raw/master/docs/screenshot.jpg)

**Comment by codernix** [comment]
lol, it rendered correctly though :)

May you share what have you found was stopping TextOFX from rendering? Was it any Natron-related stuffs? Or was it something else?

**Comment by rodlie** [comment]
Yes, but it's designed to be a backend for a WYSIWYG text editor in Natron. The plugin converts qrichtext html input to pango markup that is then rendered (see previous screenshot).

regarding TextOFX:
The font setup/generation didn't work with Fusion. The plugin was designed for Natron and uses several features only present in Natron.

**Comment by codernix** [comment]
Cool, roger that, and looking forward to your port work.

Thanks man for stepping in.

**Comment by codernix** [comment]
I haven't slept yet, so I've passed the 24 hours with no rest, and I'm yet to figure out why your code doesn't seem to work on my end :/ seems like I need to rest for a bit, then restart again.

**Comment by rodlie** [comment]
The basics seems to work now (OFX support in Fusion is not good, gives me way too many limitation), still some features missing. Will look at that in a couple of days, I'm a bit busy these next days.

**Comment by codernix** [comment]
Seems good to me. Right now, I'm not interested in like %90 of the features. I haven't really play with C/C++ for over 10 years now, but this was the opportunity to re-polish my knowledge and get to re-know  C++ anew.

It was a pleasure to see how C++ has evolved, it feels like a whole new language from how it was in a good way. This also helped me understand how OFX makes the interaction between the Client and the Host, so that's  a plus.

Back to the development, you can share just the base of your code, it would be super helpful. I mean just the part that renders the Text, no need for any other feature. That would be a good starting point to hack on.

Also, don't stress yourself about it, what you've done is already a big milestone :)

**Comment by codernix** [comment]
By any chance, is the new code supposed to be within the RichText.cpp/.h? It seems to me the case, isn't it?

**Comment by rodlie** [comment]
RichText.cpp does the text stuff, TextOFX.cpp does the OFX stuff :)

TextOFX.cpp runs the ``RichText::renderText`` function with params and returns the image buffer to the OFX host.

**Comment by codernix** [comment]
Oh, cool, then all I need is to use these files instead of the ones that come in the other repo and that's that :)

Time to compile something now, lol

**Comment by codernix** [comment]
Let me tell you my friend that the first time in Fusion's history Arabic can be rendered correctly!

https://ibb.co/mzThCgr

I will announce this on their official forums now and drop their support an email!

Meanwhile, I will track down the freezing issue after you first load Resolve then when you close it. I will check what's wrong with loading the fonts or something.

**Comment by rodlie** [comment]
That's nice. I'm not done with the plugin yet, so no need for any announcement yet.

I don't have any freezing issue anymore, that should be fixed.

**Comment by codernix** [comment]
In my modified plugin, where I removed the methods manually I have no freezes, but no rendering either, but on yours I have long freezes about 5 minutes or so. It could do with OS in use, I'm on Ubuntu Linux.

**Comment by rodlie** [comment]
I will look into it tomorrow.

**Comment by rodlie** [comment]
I'm on Slackware and CentOS.

**Comment by codernix** [comment]
Man, go get some sleep, you've done more than enough already :) It's dawn here now.
